cli/src/commands/split.rs

```diff
@@ -230,3 +242,26 @@ The remainder will be in the second commit.
     tx.finish(ui, format!("split commit {}", commit.id().hex()))?;
     Ok(())
 }
+
+/// Reads 'split.legacy-bookmark-behavior' from settings and prints a warning if
+/// the value is unset to alert the user to the behavior change.
+fn read_legacy_bookmark_behavior_setting(
+    settings: &jj_lib::settings::UserSettings,
+    ui: &Ui,
+) -> Result<bool, CommandError> {
+    match settings.get_bool("split.legacy-bookmark-behavior") {
```

Comment:
nit: you can use `.optional()?` to get `Option<bool>`.

fwiw, it might also be good to set the default value in `config/misc.toml`. Suppose this is a temporary flag, we wouldn't need an option to suppress the warning without changing the behavior.

Reply:
Thanks for the `.optional()` suggestion.

> Suppose this is a temporary flag, we wouldn't need an option to suppress the warning without changing the behavior.

I'm not sure I understand. Adding the default to `config/misc.toml` suppresses the warning. I think we want users to  see the warning and update their configs according to their preference. Are you suggesting that we print the warning until the user disables the legacy behavior? Based on the previous discussion on this PR I have a feeling that will be controversial. Maybe we can do that in a few months before actually removing the config option?

Reply:
> Are you suggesting that we print the warning until the user disables the legacy behavior?

Yes. If the warning could be suppressed by explicitly enabling the legacy behavior, the user would forget about that.

> Maybe we can do that in a few months before actually removing the config option?

That sounds also good. I would take this option if there are high chance we'll revert the deprecation.

Reply:
> Are you suggesting that we print the warning until the user disables the legacy behavior? Based on the previous discussion on this PR I have a feeling that will be controversial.

I think this is the correct path. We're just including the option to follow our convention on depending for breaking changes. If it's a significant impediment to someone's workflow, hopefully the option and warning will get them to speak up, and we can prioritize a longer-term solution like a flag to switch which commit inherits the change ID.

Reply:
That's fine with me. @ilyagr @martinvonz

Reply:
SGTM, I agree with Emily and Yuya. I'm not sure we gain much by waiting a few months.

I'm also OK with moving slower; IMO the main reason for slowing down would be if people want to keep working on this invisibly to the users. My sense is that this is not where people are at, though.

Reply:
Ok, then it's decided. We will print the warning unless the user enables the new behavior. After the new behavior becomes the default, we will print the warning if the user is opting out of the new behavior.

Reply:
Wait I’m a bit confused – I thought surely we’d use the new behaviour by default? `jj split` warning out of the box seems bad. Only users who rely on the old behaviour should have to adjust their config, right? Maybe there was a miscommunication here.